src/mscorlib/shared/System/BitConverter.cs

```diff
@@ -21,6 +21,20 @@ public static class BitConverter
 #else
         public static readonly bool IsLittleEndian = true;
 #endif
+        private static T ReadUnaligned<T>(ReadOnlySpan<byte> value) where T : struct
+        {
+            if (value.Length < Unsafe.SizeOf<T>())
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.value);
+            return Unsafe.ReadUnaligned<T>(ref value.DangerousGetPinnableReference());
+        }
+
+        private static bool WriteUnaligned<T>(T value, Span<byte> bytes) where T : struct
+        {
+            if (bytes.Length < Unsafe.SizeOf<T>())
```

Comment:
> The existing array versions do not use generics either.

Yeah, I just figured that was primarily because they were written before generics were available.

> The code may be equivalent, but you pay for the extra generic plumbing (the generic method definition, instantiation information

Ok, it can be changed.  @nchikanov, sorry for causing you extra work :)